astropy/utils/tests/test_data.py

```diff
@@ -1605,7 +1604,7 @@ def test_get_fileobj_str(a_file):
 
 def test_get_fileobj_localpath(a_file):
     fn, c = a_file
-    with get_readable_fileobj(py.path.local(fn)) as rf:
+    with get_readable_fileobj(pathlib.Path(fn)) as rf:
```

Comment:
The problem with this change is that now `test_get_fileobj_localpath()` is exactly the same as `test_get_fileobj_pathlib()`, which is the very next test in this file. Such duplication should be avoided.

[`py.path`](https://py.readthedocs.io/en/latest/path.html) provided a high-level API for filesystem paths, but it became obsolete when Python 3.4 introduced the built-in [`pathlib`](https://docs.python.org/3/library/pathlib.html) module. The question here is whether `astropy` should still explicitly check that `get_readable_fileobj()` works with the legacy `py.path`, in which case there should be a [`noqa` comment](https://beta.ruff.rs/docs/tutorial/#ignoring-errors) on this line, or whether we no longer care about `py.path` enough to have this test.

Reply:
> we no longer care about py.path enough to have this test

I am leaning towards not caring.

Reply:
Ok. Should I go ahead and remove the test all together? The PR would pretty be removing "PTH124" from the ignore list.

Reply:
I think that was what @eerovaher was eluding to but let wait to see what he says. Thanks!

Reply:
As I already said there is no point in replacing `py.path.local` with `pathlib.Path` because the latter is already being tested in `test_get_fileobj_pathlib()`. So if testing with `py.path` is no longer considered important then `test_get_fileobj_localpath()` serves no purpose and should indeed be deleted.

Reply:
I agree. Let's rm the test. Thanks!